src/components/Asset/AssetActions/Download.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import FileIcon from '@shared/FileIcon'
+import Price from '@shared/Price'
+import { useAsset } from '@context/Asset'
+import { useWeb3 } from '@context/Web3'
+import ButtonBuy from '@shared/ButtonBuy'
+import { secondsToString } from '@utils/ddo'
+import AlgorithmDatasetsListForCompute from './Compute/AlgorithmDatasetsListForCompute'
+import styles from './Download.module.css'
+import { FileMetadata, LoggerInstance } from '@oceanprotocol/lib'
+import { order } from '@utils/order'
+import { AssetExtended } from 'src/@types/AssetExtended'
+import { buyDtFromPool, calculateBuyPrice } from '@utils/pool'
+import { downloadFile } from '@utils/provider'
+import { getOrderFeedback } from '@utils/feedback'
+
+export default function Download({
+  asset,
+  file,
+  isBalanceSufficient,
+  dtBalance,
+  fileIsLoading,
+  isConsumable,
+  consumableFeedback
+}: {
+  asset: AssetExtended
+  file: FileMetadata
+  isBalanceSufficient: boolean
+  dtBalance: string
+  fileIsLoading?: boolean
+  isConsumable?: boolean
+  consumableFeedback?: string
+}): ReactElement {
+  const [accessDetails, setAccessDetails] = useState<AccessDetails>()
```

Comment:
do we specifically need to store this in a new state? I mean it's under `asset.accessDetails` so can be grabbed from there or checked from there if it's there or not